File: common/src/main/java/com/ssblur/scriptor/word/descriptor/focus/inventory/CasterFirstEmptySlotDescriptor.java
package com.ssblur.scriptor.word.descriptor.focus.inventory;

import com.ssblur.scriptor.helpers.targetable.InventoryTargetable;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.word.descriptor.focus.FocusDescriptor;
import net.minecraft.world.item.ItemStack;

public class CasterFirstEmptySlotDescriptor extends Descriptor implements FocusDescriptor {
  @Override
  public Cost cost() {
    return new Cost(0, COSTTYPE.ADDITIVE);
  }

  @Override
  public Targetable modifyFocus(Targetable targetable) {
    if(targetable instanceof InventoryTargetable inventoryTargetable)
      inventoryTargetable.useFirstMatchingSlot(ItemStack::isEmpty);
    return targetable;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/focus/inventory/CasterFirstEmptySlotDescriptor.kt
package com.ssblur.scriptor.word.descriptor.focus.inventory

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.helpers.targetable.InventoryTargetable
import com.ssblur.scriptor.helpers.targetable.Targetable
import com.ssblur.scriptor.word.descriptor.focus.FocusDescriptor
import net.minecraft.world.item.ItemStack

class CasterFirstEmptySlotDescriptor : Descriptor(), FocusDescriptor {
  override fun cost() = Cost(0.0, COSTTYPE.ADDITIVE)

  override fun modifyFocus(targetable: Targetable): Targetable {
    if (targetable is InventoryTargetable) targetable.useFirstMatchingSlot { obj: ItemStack -> obj.isEmpty }
    return targetable
  }
}
